### PR TITLE
Add --max-tablesizev-rows parameter.

### PR DESCRIPTION
**Pull request by dawidxc**
Add possibility to limit the number of rows returned from the DBC.TableSizeV table.

### CODE REVIEW

File: dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/ConnectorArguments.java [hunk 1]
```diff
@@ -131,6 +133,12 @@ public class ConnectorArguments extends DefaultArguments {
     private final OptionSpec<String> optionDatabase = parser.accepts(OPT_DATABASE, "Database(s) to export").withRequiredArg().ofType(String.class).withValuesSeparatedBy(',').describedAs("db0,db1,...");
     private final OptionSpec<String> optionSchema = parser.accepts(OPT_SCHEMA, "Schemata to export").withRequiredArg().ofType(String.class).withValuesSeparatedBy(',').describedAs("sch0,sch1,...");
     private final OptionSpec<Void> optionAssessment = parser.accepts(OPT_ASSESSMENT, "Whether to create a dump for assessment (i.e., dump additional information).");
+
+    private final OptionSpec<Long> optionMaxTableSizeVRows = parser.accepts(OPT_MAX_TABLESIZEV_ROWS,
```
Comment (frsv) [inline dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/ConnectorArguments.java:137]: You maybe also want to add an annotation to the TeradataMetadataConnector to indicate that it respects this parameter. Then this option will be shown in the connector-related documentation in `--help` output.

See `@RespectsArgumentDatabasePredicate` as an example

**Review by frsv — APPROVED** [review]

**Review by cyulysses-corp — APPROVED** [review]

File: dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/ConnectorArguments.java (hunk 1 above)
Comment (dawidxc) [reply dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/ConnectorArguments.java:137]: Done.

File: dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/ConnectorArguments.java [hunk 2]
```diff
@@ -97,6 +98,7 @@ public class ConnectorArguments extends DefaultArguments {
     public static final String OPT_DATABASE = "database";
     public static final String OPT_SCHEMA = "schema";
     public static final String OPT_ASSESSMENT = "assessment";
+    public static final String OPT_MAX_TABLESIZEV_ROWS = "max-tablesizev-rows";
```
Comment (paolomorandini) [inline dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/ConnectorArguments.java:101]: Other dialect-specific flags incorporate the dialect in their name, e.g. `OPT_ORACLE_SID` -> `--oracle-sid`.
Can we have `OPT_TERADATA` as prefix?

**Review by frsv — APPROVED** [review]

**Review by cyulysses-corp — APPROVED** [review]

File: dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/teradata/TeradataMetadataConnector.java [hunk 3]
```diff
@@ -41,6 +42,8 @@
 @Description("Dumps metadata from Teradata.")
 @RespectsArgumentDatabasePredicate
 @RespectsArgumentAssessment
+@RespectsInput(order = 450, arg = ConnectorArguments.OPT_MAX_TABLESIZEV_ROWS,
+    description = "Max number of rows to extract from DBC.TableSizeV table (available for 'teradata' connector only)")
```
[the remark this answers is not in the document]
Comment (dawidxc) [reply dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/teradata/TeradataMetadataConnector.java]: Done.

**Pull request merged**